docs/Account.md

```diff
@@ -192,23 +168,76 @@ Where:
+struct AccountCallArray:
+    member to: felt
+    member selector: felt
+    member data_offset: felt
+    member data_len: felt
+end
 ```
 
-You can read more about how messages are structured and hashed in the [Account message scheme  discussion](https://github.com/OpenZeppelin/cairo-contracts/discussions/24). For more information on the design choices and implementation of multicall, you can read the [How should Account multicall work discussion](https://github.com/OpenZeppelin/cairo-contracts/discussions/27).
+Where:
 
-> Note that the scheme of building multicall transactions within the `__execute__` method will change once StarkNet allows for pointers in struct arrays. In which case, multiple transactions can be passed to (as opposed to built within) `__execute__`.
+* `to` is the address of the target contract of the message
+* `selector` is the selector of the function to be called on the target contract
+* `data_offset` is the starting position of the calldata array that holds the `Call`'s calldata
+* `data_len` is the number of calldata elements in the `Call`
+
+## Multicall transactions
+
+A multicall transaction packs the `to`, `selector`, `calldata_offset`, and `calldata_len` of each call into the `AccountCallArray` struct and keeps the cumulative calldata for every call in a separate array. The `__execute__` function rebuilds each message by combining the `AccountCallArray` with its calldata (demarcated by the offset and calldata length specified for that particular call). The rebuilding logic is set in the internal `_from_call_array_to_call`.
+
+This is the basic flow:
+
+1. The user sends the messages for the transaction through a Signer instantiation which looks like this:
+
+    ```python
+    await signer.send_transaction(
+            account, [
+                (contract_address, 'contract_method', [arg_1]),
+                (contract_address, 'another_method', [arg_1, arg_2])
+            ]
+        )
+    ```
+
+    The `_from_call_to_call_array` method in [utils.py](../tests/utils.py) converts each call into the `AccountCallArray` format and cumulatively stores the calldata of every call into a single array. Next, both arrays (as well as the `sender`, `nonce`, and `max_fee`) are used to create the transaction hash. The Signer then invokes `__execute__` with the signature and passes `AccountCallArray`, calldata, and nonce as arguments.
+
+2. The `__execute__` method creates an empty array which is passed with the `AccountCallArray` and calldata into the `_from_call_array_to_call` method. This method iterates through the `AccountCallArray` messages and converts each message into a `Call`.
+
+3. A new empty array is created and passed with the list of calls (populated from `from_call_array_to_call`) to `_execute_list`. This method iterates and executes a contract call for each `Call` and pushes each result into the array. Once all contract calls are finished, the array holding each contract call response is returned.
```

Comment:
I think this is just an implementation detail, it's the interface what we want to document, right?

Reply:
Correct. I broke this down for myself so I could better explain/understand the process. A blog post or something might be better suited to document the flow. Good call

Reply:
I think it's a bit confusing to go that deep into this kind details, e.g. we could say "An array of `Call`s is passed to `_execute_list`" instead -- no need to explain that a new array is allocated, iterated, etc. I'd remove this section alltogether.